backend-tests/tests/test_decommission.py

```diff
@@ -0,0 +1,275 @@
+# Copyright 2020 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import pytest
+import random
+import time
+import logging
+import re
+
+from testutils.common import mongo, clean_mongo
+from testutils.api.client import ApiClient
+import testutils.api.useradm as useradm
+import testutils.api.deviceauth as deviceauth
+import testutils.api.deviceauth_v2 as deviceauth_v2
+import testutils.api.tenantadm as tenantadm
+import testutils.api.deployments as deployments
+import testutils.api.inventory as inventory
+import testutils.infra.conductor as conductor
+from testutils.infra.cli import CliTenantadm, CliUseradm, CliDeviceauth
+import testutils.util.crypto
+from testutils.common import User, Device, Tenant, \
+        create_org, create_random_authset, get_device_by_id_data, \
+        change_authset_status, create_user
+
+logging.basicConfig(format='%(asctime)s %(message)s')
+logger = logging.getLogger('test_decomission')
+#logger.setLevel(logging.INFO)
+logger.setLevel(logging.DEBUG)
+
+@pytest.yield_fixture(scope='function')
+def clean_migrated_mongo(clean_mongo):
+    deviceauth_cli = CliDeviceauth()
+    useradm_cli = CliUseradm()
+
+    deviceauth_cli.migrate()
+    useradm_cli.migrate()
+
+    yield clean_mongo
+
+@pytest.yield_fixture(scope='function')
+def clean_migrated_mongo_mt(clean_mongo):
+    deviceauth_cli = CliDeviceauth()
+    useradm_cli = CliUseradm()
+    tenantadm_cli = CliTenantadm()
+    for t in ['tenant1', 'tenant2']:
+        deviceauth_cli.migrate(t)
+        useradm_cli.migrate(t)
+        tenantadm_cli.migrate()
+
+    yield clean_mongo
+
+@pytest.yield_fixture(scope="function")
+def user(clean_migrated_mongo):
+    yield create_user('user-foo@acme.com', 'correcthorse')
+
+@pytest.yield_fixture(scope="function")
+def devices(clean_migrated_mongo, user):
+    uc = ApiClient(useradm.URL_MGMT)
```

Comment:
Honestly, I cannot figure out what `uc` stands for, Useradm Client? It would help if we follow the same convention in the test: usradmm, dvauthm, devauthd.

Reply:
Same in other places of this file, but I won't comment in all of them.

Reply:
fixed